### DISCUSSION TITLE
Customizing Virtuoso scrollbars with react-custom-scrollbars

### DISCUSSION THREAD

**Discussion by royporter7** [petyosi/react-virtuoso#743] [Q&A]
I'm trying to customize the scroll bars with the popular library react-custom-scrollbars.

I followed some basic documentation around [customizing the components](https://virtuoso.dev/customize-structure/#typescript-interfaces), however I'm continually seeing an error from Virtuoso `Cannot read properties of null (reading 'scrollTop')`.

Is there a way to make this integration work?

Basic sandbox attempting to integrate react-custom-scrollbars and Virtuoso:
https://codesandbox.io/s/distracted-jasper-xe3sct?file=/src/App.js

**Comment by lexahall** [comment]
I'm also getting this error, haven't had any luck so far.

**Reply by royporter7** [reply]
Me either - glad to hear that I'm not alone in not being able to wrangle react-custom-scrollbars and Virtuoso together.

If a solution was ever shared I would implement it in a heart beat - for now I just have some overly default looking scrollbars.

**Reply by lexahall** [reply]
I was able to fix my situation, and it turned out that the ref I was passing wasn't defined. I think the `Cannot read properties of null (reading 'scrollTop')` is happening when react-virtuoso has a null ref to your Scrollable. 

I looked at your sandbox a little bit, but I'm not familiar with react-custom-scrollbars and wasn't immediately able to see a solution. I have a hunch that passing the ref into another callback function (renderView) might be an issue here? 

I also noticed in the docs that you should always pass through the props for a renderView call, like so: 
![image](https://user-images.githubusercontent.com/10774982/196565889-2f6511e9-f80f-4f66-89a8-cdff71c49423.png)

**Reply by paulm17** [reply]
@lexahall Any chance you could make a working sandbox?  Thanks

**Reply by lexahall** [reply]
Here is a really simple example with just a default custom scroller https://codesandbox.io/s/eager-sanderson-0vxy20?file=/src/App.js

I hope this is helpful!

**Comment by zoy-l** [comment]
```js
const animationCount = Array.from(Array(101), (_, i) => i)

const TackColorAnimationEnd = animationCount.reduce(
  (memo, current) => {
    memo['@keyframes TackColor-Animation-End'][`${current}%`] = {
      '--CustomScrollBar-TackColor': `rgba(var(--CustomScrollBar-TackColor-Base),${
        (100 - current) / 100
      })`
    }
    return memo
  },
  {
    '@keyframes TackColor-Animation-End': {} as AnyObject
  }
)

const TackColorAnimationStart = animationCount.reduce(
  (memo, current) => {
    memo['@keyframes TackColor-Animation-Start'][`${current}%`] = {
      '--CustomScrollBar-TackColor': `rgba(var(--CustomScrollBar-TackColor-Base),${current / 100})`
    }
    return memo
  },
  {
    '@keyframes TackColor-Animation-Start': {} as AnyObject
  }
)

const CustomScrollBar = {
  ...TackColorAnimationEnd,
  ...TackColorAnimationStart,
  '.CustomScrollBar': {
    overflowX: 'hidden',
    overflowY: 'overlay !important',
    '--CustomScrollBar-TackColor': 'transparent',
    animation: 'TackColor-Animation-End 0.3s',
    animationFillMode: 'forwards',

    '&::-webkit-scrollbar-thumb': {
      backgroundColor: 'var(--CustomScrollBar-TackColor)',
      borderRadius: 3
    },

    '&:hover': {
      animation: 'TackColor-Animation-Start 0.3s',
      animationFillMode: 'forwards'
    },

    '&::-webkit-scrollbar-track': {
      backgroundColor: 'transparent'
    },

    '&::-webkit-scrollbar': {
      width: 6,
      backgroundColor: 'transparent'
    }
  }
}
```
one solution I use in `electron` can get perfect scroll bars
